src/providers/twitch/twitchapi.cpp

```diff
@@ -0,0 +1,37 @@
+#include "providers/twitch/twitchapi.hpp"
+
+#include "Application.hpp"
+#include "common/UrlFetch.hpp"
+#include "controllers/accounts/AccountController.hpp"
+#include "providers/twitch/TwitchCommon.hpp"
+
+#include <QString>
+#include <QThread>
+
+namespace chatterino {
+
+void TwitchApi::FindUserId(const QString user, std::function<void(QString)> callback)
+{
+    QString requestUrl("https://api.twitch.tv/kraken/users?login=" + user +
+                       "&api_version=5&client_id=" + getDefaultClientID());
+    NetworkRequest request(requestUrl);
+    request.setCaller(QThread::currentThread());
+    request.setTimeout(30000);
+    request.onSuccess([callback](auto result) mutable {
```

Comment:
If Twitch were to change the structure of this API somehow, or return invalid data, it could potentially cause a crash.
See how it's done in https://github.com/fourtf/chatterino2/blob/master/src/providers/twitch/PartialTwitchUser.cpp#L40